examples/ota-requestor-app/esp32/main/OTARequesterImpl.cpp

```diff
@@ -0,0 +1,311 @@
+        constexpr EmberAfOTADownloadProtocol kExampleProtocolsSupported[] = { EMBER_ZCL_OTA_DOWNLOAD_PROTOCOL_BDX_SYNCHRONOUS };
+        const char locationBuf[]                                          = { 'U', 'S' };
+        CharSpan exampleLocation(locationBuf);
+        constexpr bool kExampleClientCanConsent = false;
+        ByteSpan metadata;
+
+        QueryImage::Type args;
+        args.vendorId           = kExampleVendorId;
+        args.productId          = kExampleProductId;
+        args.softwareVersion    = kExampleSoftwareVersion;
+        args.protocolsSupported = kExampleProtocolsSupported;
+        args.hardwareVersion.Emplace(kExampleHWVersion);
+        args.location.Emplace(exampleLocation);
+        args.requestorCanConsent.Emplace(kExampleClientCanConsent);
+        args.metadataForProvider.Emplace(metadata);
+
+        err = cluster.InvokeCommand(args, nullptr, OnQueryImageResponse, OnQueryImageFailure);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(SoftwareUpdate, "QueryImage() failed: %" CHIP_ERROR_FORMAT, err.Format());
+        }
+        break;
+    }
+    case kCommandApplyUpdateRequest: {
+        constexpr uint32_t kNewVersion = 1;
+
+        ApplyUpdateRequest::Type args;
+        args.updateToken = ByteSpan(mOtaUpdateToken, mOtaUpdateTokenLen);
+        args.newVersion  = kNewVersion;
+
+        err = cluster.InvokeCommand(args, nullptr, OnApplyUpdateResponse, OnApplyUpdateRequestFailure);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(SoftwareUpdate, "ApplyUpdateRequest() failed: %s", chip::ErrorStr(err));
+        }
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void OnConnectionFailure(void * context, NodeId deviceId, CHIP_ERROR error)
+{
+    ChipLogError(SoftwareUpdate, "failed to connect to 0x%" PRIX64 ": %" CHIP_ERROR_FORMAT, deviceId, error.Format());
+}
+
+void OnBlockReceived(void * context, const chip::bdx::TransferSession::BlockData & blockdata)
+{
+    if (OTAUpdater::GetInstance().IsInProgress() == false)
+    {
+        OTAUpdater::GetInstance().Begin();
+    }
+    // TODO: Process/skip the Matter OTA header
+    OTAUpdater::GetInstance().Write(reinterpret_cast<const void *>(blockdata.Data), blockdata.Length);
+}
+
+void OnTransferComplete(void * context)
+{
+    ESP_LOGI(TAG, "Transfer complete!");
+    OTAUpdater::GetInstance().End();
+}
+
+void OnTransferFailed(void * context, BdxDownloaderErrorTypes status)
+{
+    ESP_LOGI(TAG, "Transfer Failed, status:%x", status);
+    OTAUpdater::GetInstance().Abort();
+}
+
+void ConnectToProvider(const char * ipAddress, uint32_t nodeId)
+{
+    NodeId providerNodeId = nodeId;
+
+    chip::OperationalDeviceProxy * operationalDeviceProxy = Server::GetInstance().GetOperationalDeviceProxy();
+    if (operationalDeviceProxy != nullptr && operationalDeviceProxy->GetDeviceId() != providerNodeId)
+    {
+        operationalDeviceProxy->Disconnect();
```

Comment:
```suggestion
        operationalDeviceProxy->Disconnect();
        delete operationalDeviceProxy;
```
The object is created with `new` before. Please also consider using static variables or `chip::Platform::New`.